src/local-development/ask-origin.ts

```diff
@@ -29,39 +33,69 @@ export async function askForOrigin(
 	origins: LocalAppOrigin[],
 	makeappRootdir: vscode.Uri,
 	purposeLabel?: string,
+	enableFeatureAddNewOrigin: boolean = false,
 ): Promise<LocalAppOriginWithSecret | undefined> {
-	return includeApiKey(await askForOrigin2(origins, purposeLabel), makeappRootdir);
+	return includeApiKey(
+		await askForOrigin2(origins, makeappRootdir, purposeLabel, enableFeatureAddNewOrigin),
+		makeappRootdir,
+	);
 }
 
-async function askForOrigin2(origins: LocalAppOrigin[], purposeLabel?: string): Promise<LocalAppOrigin | undefined> {
+async function askForOrigin2(
+	origins: LocalAppOrigin[],
+	makeappRootdir: vscode.Uri,
+	purposeLabel?: string,
+	enableFeatureAddNewOrigin: boolean = false,
+): Promise<LocalAppOrigin | undefined> {
 	if (!Array.isArray(origins)) {
 		throw new Error('Origins should be the array.');
 	}
 
```

Comment:
These cases could use some comments about their purpose to make it less dirty 🙏

Reply:
Thanks for the suggestion. You are right, added, improved.